### PR TITLE
Remove deleted pipelines from global active pipeline list

### LINKED ISSUE #29: Delete pipeline needs manual restart
The only way to delete a pipeline is to manually remove the binary in the pipeline folder. 
Then a manual restart of gaia is required since the pipeline is not automatically removed from the [active global pipeline list](https://github.com/gaia-pipeline/gaia/blob/master/pipeline/pipeline.go#L50).

What we have to do:
During the [pipeline check](https://github.com/gaia-pipeline/gaia/blob/master/pipeline/ticker.go#L60) we should check if a pipeline has been removed. If so, we should remove it from the active pipeline list.

### PR DESCRIPTION
**Pull request by bidhan-a**
Fixes #29 

Let me know your thoughts @michelvocks 

Thanks!

### CODE REVIEW

File: pipeline/pipeline.go [hunk 1]
```diff
@@ -157,6 +165,23 @@ func (ap *ActivePipelines) Contains(n string) bool {
 	return foundPipeline
 }
 
+// RemoveDeletedPipelines removes the pipelines whose names are NOT
+// present in `existingPipelineNames` from the given ActivePipelines instance.
+func (ap *ActivePipelines) RemoveDeletedPipelines(existingPipelineNames []string) {
+	for i, pipeline := range ap.Pipelines {
```
[the remark this answers is not in the document]
Comment (bidhan-a) [reply pipeline/pipeline.go]: @michelvocks The iter function doesn't provide a counter so we'd have to introduce a counter variable to get the pipeline index. Would this be okay?

[the remark this answers is not in the document]
Comment (michelvocks) [reply pipeline/pipeline.go]: @bidhan-a Yes that is fine. :smile:

[the remark this answers is not in the document]
Comment (bidhan-a) [reply pipeline/pipeline.go]: @michelvocks Cool! I'll implement that and let your know.

**Comment by bidhan-a** [comment]
Hey @michelvocks 

I have encountered an issue.

We're using an RLock inside the Iter() function: https://github.com/gaia-pipeline/gaia/blob/master/pipeline/pipeline.go#L136

Similarly, I have added a Lock inside the Remove() function: https://github.com/bidhan-a/gaia/blob/issue-29/pipeline/pipeline.go#L92

If we try to remove pipelines while Iter() is running (https://github.com/bidhan-a/gaia/blob/issue-29/pipeline/pipeline.go#L181), then `Remove` will try to lock an already locked resource and the code will not move further. 

Any way we can circumvent this?

Thanks

**Comment by michelvocks** [comment]
Good point @bidhan-a. I think the only viable way to work around this is to copy all elements which you want to delete into an intermediate slice. After you have left the `Iter()` loop you can loop through the intermediate slice and safely remove the items. What do you think? 🤗

**Comment by bidhan-a** [comment]
Sounds good to me @michelvocks !

**Comment by bidhan-a** [comment]
All done @michelvocks. I've also included tests for the functions which I've added.

Thanks!

**Pull request merged**